pkg/web_app/lib/src/_dom_helper.dart

```diff
@@ -135,19 +135,31 @@ Future<bool> modalConfirm(Element content) async {
 /// false, it will omit the "Cancel" button.
 ///
 /// Returns true if "OK" was pressed, false otherwise.
-Future<bool> _modalWindow({
+Future<bool> modalWindow({
   @required String titleText,
   @required Element content,
   @required bool isQuestion,
   String cancelButtonText,
   String okButtonText,
+
+  /// If specified, this callback function will be called on "OK" button clicks,
+  /// and the dialog window will be kept open. Returns with the success status
+  /// of the operation, and if `true`, the dialog will be closed, otherwise the
+  /// dialog is kept open (e.g. the user may update the form).
+  FutureOr<bool> Function() onOkButtonClicked,
```

Comment:
Hmm... so RPC is running while the dialog is open... that means that `await modalMessage('Success', successMessage);` is also displayed on top of the creation dialog.

Hmm, could we call this `execute` or some other name?